File: TrackIt/Core Data/Entry.m
//
//  Entry.m
//  TrackIt
//
//  Created by Jason Ji on 11/2/15.
//  Copyright © 2015 Jason Ji. All rights reserved.
//

#import "Entry.h"

@implementation Entry

+(instancetype)entryWithAmount:(NSNumber *)amount note:(NSString *)note date:(NSDate *)date inManagedObjectContext:(nonnull NSManagedObjectContext *)context {
    Entry *entry = [NSEntityDescription insertNewObjectForEntityForName:NSStringFromClass([self class]) inManagedObjectContext:context];
    entry.amount = amount;
    entry.note = note;
    entry.date = date;
    return entry;
}

@end

File: TrackIt/Core Data/Entry.swift
//
//  Entry.swift
//  TrackIt
//
//  Created by Jason Ji on 4/25/16.
//  Copyright © 2016 Jason Ji. All rights reserved.
//

import Foundation
import CoreData

@objc(Entry)
class Entry: NSManagedObject {

// Insert code here to add functionality to your managed object subclass

    class func entry(withAmount amount: NSNumber?, note: String?, date: Date?, tags: [Tag]?, inManagedObjectContext managedObjectContext: NSManagedObjectContext) -> Entry {
        let entry = NSEntityDescription.insertNewObject(forEntityName: "Entry", into: managedObjectContext) as! Entry
        
        entry.amount = amount
        entry.note = note
        entry.date = date
        if let tags = tags {
            entry.tags = NSSet(array: tags)
        }
        
        return entry
    }
}
